src/vivarium_csu_ltbi/data/gbd2019_hh_tb_model.py

```diff
@@ -0,0 +1,148 @@
+    prev_actb['year'] = 2019
+    prev_actb['age_group_start'] = prev_actb.age_group_name.map(lambda x: int(x.split('_to_')[0]))
+    prev_actb['age_group_end'] = prev_actb.age_group_name.map(lambda x: int(x.split('_to_')[1]))
+    prev_actb.drop(columns=['location_id', 'age_group_id', 'age_group_name', 'sex_id', 'year_id'], inplace=True)
+    prev_actb_long = prev_actb.melt(id_vars=['location', 'age_group_start', 'age_group_end', 'sex', 'year'], var_name=['draw'])
+    prev_actb_long['draw'] = prev_actb_long.draw.map(lambda x: int(x.split('_')[1]))
+    return prev_actb_long
+
+def interpolation(prev_actb: pd.DataFrame, hh_data: pd.DataFrame, draw: int):
+    """assign the probability of active TB for each household member"""
+    mask = hh_data.copy()
+    interp = Interpolation(prev_actb.query(f'draw == {draw}'),
+                           categorical_parameters=['sex'],
+                           continuous_parameters=[['age', 'age_group_start', 'age_group_end']],
+                           order=0,
+                           extrapolate=True,
+                           validate=True)
+    mask['pr_actb'] = interp(mask).value
+    return mask
+
+def calc_pr_no_actb_in_hh(df: pd.DataFrame):
+    """compute the probability that no person with active TB in a single household"""
+    pr_no_tb = 1 - df.pr_actb
+    pr_no_tb_in_hh = np.prod(pr_no_tb)
+    return pr_no_tb_in_hh
+
+def calc_age_sex_specific_no_actb_prop(df: pd.DataFrame):
+    """
+    calculate the probability of no active TB case in the household
+    for certain age and sex group
+    """
+    age_bins = [0, 1] + list(range(5, 96, 5)) + [125]
+    result = pd.DataFrame()
+    for i, age_group_start in enumerate(age_bins[:-1]):
+        age_group_end = age_bins[i+1]
+        for sex in ['Male', 'Female']:
+            hh_with = df.query(f'age >= {age_group_start} and age < {age_group_end} and sex == "{sex}"').hh_id.unique()
+            prop = df[df.hh_id.isin(hh_with)].groupby('hh_id').apply(calc_pr_no_actb_in_hh)
+            prop_mean = prop.mean()
+            age_sex_result = pd.DataFrame({'age_group_start': [age_group_start], 
+                                           'age_group_end': [age_group_end], 
+                                           'sex': [sex], 
+                                           'value': [prop_mean]})
+            result = pd.concat([result, age_sex_result], ignore_index=True)
+    return result
+
+def get_estimates(location_id: int, draw: int):
+    """calculate the prevalence of household contact"""
+    cols = ['age_group_start', 'age_group_end', 'sex']
+    df_hh = load_hh_data(location_id)
+    hh_ids = df_hh.hh_id.unique()
+    # boostrap HH data by resampling hh_id with replacement
+    sample_hhids = np.random.choice(hh_ids, size=len(hh_ids), replace=True)
+    df_hh = df_hh.set_index("hh_id")
+    df_hh_sample = df_hh.loc[sample_hhids].reset_index()
+    prev_actb = format_actb_prevalence(pull_actb_prevalence(location_id))

```

Comment:
`line 137` slow things down, we may want to update the vivarium artifact for active TB prevalence in order to pull GBD 2019 data. @collijk Our artifact seems promising for getting single draw data, right?